pkgs/by-name/ne/neovim-unwrapped/package.nix

```diff
@@ -1,70 +1,101 @@
-{ lib, stdenv, fetchFromGitHub, removeReferencesTo, cmake, gettext, msgpack-c, libiconv
-, libuv, lua, pkg-config
-, unibilium
-, libvterm-neovim
-, tree-sitter
-, fetchurl
-, buildPackages
-, treesitter-parsers ? import ./treesitter-parsers.nix { inherit fetchurl; }
-, CoreServices
-, fixDarwinDylibNames
-, glibcLocales ? null, procps ? null
-
-# now defaults to false because some tests can be flaky (clipboard etc), see
-# also: https://github.com/neovim/neovim/issues/16233
-, nodejs ? null, fish ? null, python3 ? null
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  removeReferencesTo,
+  cmake,
+  gettext,
+  msgpack-c,
+  libiconv,
+  libuv,
+  lua,
+  pkg-config,
+  unibilium,
+  libvterm-neovim,
+  tree-sitter,
+  fetchurl,
+  buildPackages,
+  treesitter-parsers ? import ./treesitter-parsers.nix { inherit fetchurl; },
+  CoreServices,
+  fixDarwinDylibNames,
+  glibcLocales ? null,
+  procps ? null,
+
+  # now defaults to false because some tests can be flaky (clipboard etc), see
+  # also: https://github.com/neovim/neovim/issues/16233
+  nodejs ? null,
+  fish ? null,
+  python3 ? null,
 }:
-stdenv.mkDerivation (finalAttrs:
+stdenv.mkDerivation (
+  finalAttrs:
   let
-  nvim-lpeg-dylib = luapkgs: if stdenv.hostPlatform.isDarwin
-    then (luapkgs.lpeg.overrideAttrs (oa: {
-      preConfigure = ''
-        # neovim wants clang .dylib
-        sed -i makefile -e "s/CC = gcc/CC = clang/"
-        sed -i makefile -e "s/-bundle/-dynamiclib/"
-      '';
-      preBuild = ''
-        # there seems to be implicit calls to Makefile from luarocks, we need to
-        # add a stage to build our dylib
-        make macosx
-        mkdir -p $out/lib
-        mv lpeg.so $out/lib/lpeg.dylib
-      '';
-      nativeBuildInputs =
-        oa.nativeBuildInputs
-        ++ (
-          lib.optional stdenv.hostPlatform.isDarwin fixDarwinDylibNames
-        );
-    }))
-    else luapkgs.lpeg;
-  requiredLuaPkgs = ps: (with ps; [
-    (nvim-lpeg-dylib ps)
-    luabitop
-    mpack
-  ] ++ lib.optionals finalAttrs.finalPackage.doCheck [
-    luv
-    coxpcall
-    busted
-    luafilesystem
-    penlight
-    inspect
-  ]
-  );
-  neovimLuaEnv = lua.withPackages requiredLuaPkgs;
-  neovimLuaEnvOnBuild = lua.luaOnBuild.withPackages requiredLuaPkgs;
-  codegenLua =
-    if lua.luaOnBuild.pkgs.isLuaJIT
-      then
-        let deterministicLuajit =
-          lua.luaOnBuild.override {
+    nvim-lpeg-dylib =
+      luapkgs:
+      if stdenv.hostPlatform.isDarwin then
+        let
+          luaLibDir = "$out/lib/lua/${lib.versions.majorMinor luapkgs.lua.luaversion}";
+        in
+        (luapkgs.lpeg.overrideAttrs (oa: {
+          preConfigure = ''
+            # neovim wants clang .dylib
+            sed -i makefile -e "s/CC = gcc/CC = clang/"
```

Comment:
Can we use substituteInPlace --replace-fail ?

Reply:
Good idea. It still looks like it builds fine.
Btw, I don't think that the last substitution (removing the `linux` make target) is necessary as we run `make macosx` anyway in the `preBuild` anyway.
Can you enlighten us on this @hurricanehrndz ?

Reply:
This is the recursive diff between derivation A obtained with:
```nix
preConfigure = ''
  sed -i makefile -e "s/CC = gcc/CC = clang/"
  sed -i makefile -e "s/-bundle/-dynamiclib/"
  sed -i makefile -e "s/lpeg.so/lpeg.dylib/"
  sed -i makefile -e '/^linux:$/ {N; d;}'
'';
```
and derivation B obtained with:
```nix
preConfigure = ''
  substituteInPlace Makefile \
    --replace-fail "CC = gcc" "CC = clang" \
    --replace-fail "-bundle" "-dynamiclib" \
    --replace-fail "lpeg.so" "lpeg.dylib"
'';
```

```
❮ diff -r /nix/store/xidrlc8ni9jmxaw95bvfr8rjh3ylc7p5-luajit2.1-lpeg-1.1.0-1 /nix/store/vsn8wlr6i6zgy7h41gb567xm0n187fzg-luajit2.1-lpeg-1.1.0-1
Binary files /nix/store/xidrlc8ni9jmxaw95bvfr8rjh3ylc7p5-luajit2.1-lpeg-1.1.0-1/lib/lua/5.1/lpeg.dylib and /nix/store/vsn8wlr6i6zgy7h41gb567xm0n187fzg-luajit2.1-lpeg-1.1.0-1/lib/lua/5.1/lpeg.dylib differ
diff '--color=auto' -r /nix/store/xidrlc8ni9jmxaw95bvfr8rjh3ylc7p5-luajit2.1-lpeg-1.1.0-1/nix-support/propagated-build-inputs /nix/store/vsn8wlr6i6zgy7h41gb567xm0n187fzg-luajit2.1-lpeg-1.1.0-1/nix-support/propagated-build-inputs
1c1
< /nix/store/m8nihfxqggb2zgvvd7nd7rmd8rb3y52h-luajit-2.1.1713773202 
\ No newline at end of file
---
> /nix/store/3y1m37q82iaqkbr02rp15a1ynxdrfvnw-luajit-2.1.1713773202 
\ No newline at end of file
```

Reply:
I guess that this is not too important, because building twice `lpeg` with the same `makefile` gives two different binaries.
So I guess that it is fine to go with this new `substituteInPlace`-only implem.

Reply:
Actually the makefile is really only used for Darwin, see previous PRs to this file. The linux targets are actually artifacts from the rockspec. Hence why we have to remove `lpeg.so` 

So yeah ` sed -i makefile -e '/^linux:$/ {N; d;}'` this is not really required.